conans/model/new_build_info.py

```diff
@@ -0,0 +1,172 @@
+import copy
+import os
+from collections import OrderedDict
+
+from conans.errors import ConanException
+from conans.model.build_info import DefaultOrderedDict
+
+_DIRS_VAR_NAMES = ["includedirs", "srcdirs", "libdirs", "resdirs", "bindirs", "builddirs",
+                  "frameworkdirs"]
+_FIELD_VAR_NAMES = ["system_libs", "frameworks", "libs", "defines", "cflags", "cxxflags",
+                   "sharedlinkflags", "exelinkflags"]
+
+
+class _BaseNewCppInfo(object):
+
+    def __init__(self):
+        self.names = {}
+        self.build_modules = {}
+
+        # ###### DIRECTORIES
+        self.includedirs = []  # Ordered list of include paths
+        self.srcdirs = []  # Ordered list of source paths
+        self.libdirs = []  # Directories to find libraries
+        self.resdirs = []  # Directories to find resources, data, etc
+        self.bindirs = []  # Directories to find executables and shared libs
+        self.builddirs = []
+        self.frameworkdirs = []
+
+        # ##### FIELDS
+        self.system_libs = []  # Ordered list of system libraries
+        self.frameworks = []  # Macos .framework
+        self.libs = []  # The libs to link against
+        self.defines = []  # preprocessor definitions
+        self.cflags = []  # pure C flags
+        self.cxxflags = []  # C++ compilation flags
+        self.sharedlinkflags = []  # linker flags
+        self.exelinkflags = []  # linker flags
+
+        self.sysroot = ""
+        self.requires = []
+
+        # ##### PATTERNS
```

Comment:
Note: This is not necessary for this PR so I will remove it and when we move this forward will add it to the layouts PR.